### PR TITLE
Allow file_type option in drop_channels

### PR DESCRIPTION
**Pull request by aklos**
drop_channels forces the target to EDF+ but it should allow you to choose the file type.

### CODE REVIEW

**Comment by skjerns** [comment]
This should already work - just set `edf_target` to end in `.bdf` and it should automatically be saved with BDF+

However, this change should be fine as well.

**Comment by aklos** [comment]
> This should already work - just set `edf_target` to end in `.bdf` and it should automatically be saved with BDF+
> 
> However, this change should be fine as well.

The issue is that I actually need a plain EDF (no annotations) to work with wfdb, so inferring from file extension isn't enough.

I'll make the requested changes, thanks!

**Comment by skjerns** [comment]
right! so vanilla EDF. Then this check makes sense. I would not have thought that vanilla EDF is still used anywhere.

Does `pyedflib` create valid vanilla EDF files? I do remember there to be some problems

**Comment by aklos** [comment]
They're valid enough that wfdb can process them. I suppose wfdb 4.0.0 now handles EDF/EDF+ without conversion, but a lot of algorithms still rely on using edf2mit which only accepts vanilla EDFs.

I'm very much a layman when it comes to this stuff. Just trying to get some heartbeat classification models working correctly and wfdb is really hard to understand.

File: pyedflib/highlevel.py [hunk 1]
```diff
@@ -441,6 +441,10 @@ def write_edf(edf_file, signals, signal_headers, header=None, digital=False,
         'file_type must be in range -1, 3'
     assert block_size<=60 and block_size>=-1 and block_size!=0, \
         'blocksize must be smaller or equal to 60'
+    if file_type != -1:
```
Comment (aklos) [inline pyedflib/highlevel.py:444]: Added a check here in write_edf to make sure file extension matches file_type opt.

**Comment by skjerns** [comment]
looks good, thanks a lot!

**Pull request merged**